pkg/cloudproviders/aws/context.go

```diff
@@ -0,0 +1,89 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
```

Comment:
Many paths/aliases above collapse if this lives in the same package as cloudprovider.go. It's the same dependency closure, anyways.

`/cloudproviders/aws/context.go`?
`/cloudproviders/aws/cloudprovider/context.go`?

In either case, we'll eventually strip the `/cloudproviders/aws` prefix once the karpenter-core migration is complete.

Reply:
It won't be the same dependency closure when we introduce `controllers/`. Maybe putting `context` into its own package is a better idea?